src/PowerShellEditorServices/Services/DebugAdapter/Handlers/ConfigurationDoneHandler.cs

```diff
@@ -103,26 +118,81 @@ private async Task LaunchScriptAsync(string scriptToLaunch)
                     // This seems to be the simplest way to invoke a script block (which contains breakpoint information) via the PowerShell API.
                     var cmd = new PSCommand().AddScript(". $args[0]").AddArgument(ast.GetScriptBlock());
                     await _executionService
-                        .ExecutePSCommandAsync<object>(cmd, new PowerShellExecutionOptions { WriteOutputToHost = true }, CancellationToken.None)
+                        .ExecutePSCommandAsync<object>(cmd, CancellationToken.None, s_debuggerExecutionOptions)
                         .ConfigureAwait(false);
                 }
                 else
                 {
                     await _executionService
                         .ExecutePSCommandAsync(
                             new PSCommand().AddScript(untitledScript.Contents),
-                            new PowerShellExecutionOptions { WriteOutputToHost = true },
-                            CancellationToken.None)
+                            CancellationToken.None,
+                            s_debuggerExecutionOptions)
                         .ConfigureAwait(false);
                 }
             }
             else
             {
-                //await _powerShellContextService
-                //    .ExecuteScriptWithArgsAsync(scriptToLaunch, _debugStateService.Arguments, writeInputToHost: true).ConfigureAwait(false);
+                await _executionService
+                    .ExecutePSCommandAsync(
+                        BuildPSCommandFromArguments(scriptToLaunch, _debugStateService.Arguments),
+                        CancellationToken.None,
+                        s_debuggerExecutionOptions)
+                    .ConfigureAwait(false);
             }
 
             _debugAdapterServer.SendNotification(EventNames.Terminated);
         }
+
+        private PSCommand BuildPSCommandFromArguments(string command, IReadOnlyList<string> arguments)
+        {
+            if (arguments is null
+                || arguments.Count == 0)
+            {
+                return new PSCommand().AddCommand(command);
+            }
+
+            // We are forced to use a hack here so that we can reuse PowerShell's parameter binding
+            var sb = new StringBuilder()
```

Comment:
Ah in this case I prefer `var` for variables. I discussed it with some others on the team like @daxian-dbw and we all agreed that `new()` makes sense for fields but that we prefer `var` for variables. For different reasons, but in my case because the type of the rvalue is the ground truth and belongs on the RHS.